src/component/Tooltip.tsx

```diff
@@ -110,16 +114,16 @@ function TooltipInternal<TValue extends ValueType, TName extends NameType>(props
     wrapperStyle,
     cursor,
     shared,
+    trigger,
     portal: portalFromProps,
   } = props;
   const viewBox = useViewBox();
   const accessibilityLayer = useAccessibilityLayer();
-  const { active: activeFromContext, payload: payloadFromProps, coordinate, label } = useTooltipContext();
-  // TODO this will fail tests if we use the selector, uncomment and fix
-  // const payloadFromContext = useAppSelector(state => selectTooltipPayload(state, shared));
-  // const payload = payloadFromContext?.length > 0 ? payloadFromContext : payloadFromProps;
-  const payload = payloadFromProps;
   const tooltipEventType = useTooltipEventType(shared);
+  const { active: activeFromContext, payload: payloadFromProps, coordinate, label } = useTooltipContext();
+  const payloadFromContext = useAppSelector(state => selectTooltipPayload(state, tooltipEventType, trigger));
```

Comment:
I am trying to avoid pushing all of Tooltip props to Redux state - but I am afraid that the existence of `defaultIndex` will force me to do that.